### ISSUE TITLE
BaseReferenceDrawer constant uses wrong height for a custom struct

### ISSUE THREAD

**Issue by Robert-Oost** [DanielEverland/ScriptableObject-Architecture#119]
### Issue
After generating a `Reference` ScriptableObject script with the SO code generator using a custom serializable struct with its own propertyDrawer, the representation of the struct has the wrong height when folded out in the editor. As far as I know, this only happens for SO References, not any of the other SO classes. It requires the "Use Constant" flag to be true. The height for the resulting representation of the property is always wrong, as the foldout icon takes up an entire line's height.

![image](https://user-images.githubusercontent.com/39600335/78368951-bd687200-75c4-11ea-9685-f01b8002ecd0.png)

![image](https://user-images.githubusercontent.com/39600335/78367172-f05d3680-75c1-11ea-8267-8b42a5c95a2b.png)

### Reproduction flow
Take a simple struct, in this case called `CameraView`. It contains two Vector3 values, to determine the desired rotation (in Euler angles) and position of a camera. This is the struct definition:
```
[Serializable]
public struct CameraView {
    public Vector3 cameraPosition;
    public Vector3 cameraRotation;
}
```
Since the struct is serializable by the UnityEditor, we have a very basic PropertyDrawer, that just draws the fields as intended:

**OnGUI:** 
```
EditorGUI.PropertyField(position, property, label, true);
```
And **GetPropertyHeight**:
```
return EditorGUI.GetPropertyHeight(property);
```

After this, we generate SO Architecture code for the `CameraView` type, including a `CameraViewReference`. This allows us to create a few `CameraViewVariable` assets, which look fine in the editor:
![image](https://user-images.githubusercontent.com/39600335/78368023-47174000-75c3-11ea-9afa-4629f0632457.png)
_(Of course, we're ignoring the developer description as that was already mentioned in a different issue.)_ 

When we finally include this reference in a MonoBehaviour, like so:
```
public CameraViewReference cameraView;
```
It looks fine as a variable:
![image](https://user-images.githubusercontent.com/39600335/78369452-821a7300-75c5-11ea-94f5-c5c0859b10ba.png)

But when we select "Use Constant", we are immediately presented with the foldout icon, positioned a bit strangely:
![image](https://user-images.githubusercontent.com/39600335/78369574-afffb780-75c5-11ea-9684-25f0ea29f80f.png)
This isn't a major issue though, as it's still possible to click the rightmost edge of the foldout icon, which results in the images shown at the top of this issue.

Another issue would be the indentation of this struct, as it would be better suited using the full width underneath. However, that is not the main focus.

### Potential solution
After digging around in the `BaseReferenceDrawer`, I found a potential solution. In `GetPropertyHeight`, the `SupportsMultiLine` bool is false for custom structs. When manually set to true for testing purposes, the height changes when the field is folded out.
Closed:
![image](https://user-images.githubusercontent.com/39600335/78370984-ac6d3000-75c7-11ea-9350-3161ac13068b.png)
Folded out:
![image](https://user-images.githubusercontent.com/39600335/78371053-c149c380-75c7-11ea-9aa9-953c6ea5a1b7.png)

But as you can see, the height is still a bit strange, as it now results in a bit of unused space. Regardless, I noticed the use of the [MultiLine] attribute for classes would allow one to make use of this behaviour. However, as it stands, this is only reserved for classes, and not for structs.

**Comment by DanielEverland** [comment]
Thanks for reporting this issue in so much detail!

I should hopefully have some time in the coming week to catch up with the increasing list of issues :)

**Comment by Robert-Oost** [comment]
I've encountered the same problem in the collection editor.
![image](https://user-images.githubusercontent.com/39600335/79354654-f5fd3980-7f3c-11ea-85f7-3e5f2830249f.png)

I did things a bit different this time along. I generated these scriptable objects for a class instead of a struct, and putting the MultiLine attribute above the class declaration doesn't seem to have any effect.

I figure that this is probably the result of the same problem. To reproduce this one, just follow the original reproduction flow, but generate a collection as well. Using a class VS a struct might make a difference but I'm not completely sure about that.

**Comment by DanielEverland** [comment]
I'm working on a fix by implementing #110 - this is, as you might imagine, quite a big feature. Once it's done everything should render properly, since I'm also re-writing all the editors.

Once it's done you won't have to implement PropertyDrawers either, unless you want some custom behaviour, of course.

**Comment by DanielEverland** [comment]
This will be fixed in the new update. Just added the finishing touches.

![](https://i.imgur.com/1IP6D9q.gif)

This update includes drawing custom objects without property drawers. The above is an example of that, the type simply including the following:

```csharp
[System.Serializable]
public class TestType
{
    [SerializeField]
    private float floatValue;
    [SerializeField]
    private Quaternion quaternionValue;
    [SerializeField]
    private SubType subType;

    [System.Serializable]
    public class SubType
    {
        public double publicValue;

        [SerializeField]
        private string privateValue;
    }
}
```
